cloudwatch_to_firehose_role.tf

```diff
@@ -0,0 +1,68 @@
+# ------------------------------------------------------------------------------
+# Provision the IAM resources that will allow CloudWatch to send log events to a
+# Firehose delivery stream.
+# ------------------------------------------------------------------------------
+
+# Create an IAM policy document that allows CloudWatch to assume the role that
+# this policy is attached to.
+data "aws_iam_policy_document" "cloudwatch_assume_role" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+      "sts:TagSession",
+    ]
+
+    effect = "Allow"
+
+    principals {
+      identifiers = ["logs.amazonaws.com"]
+      type        = "Service"
+    }
+  }
+}
+
+# Create the IAM role that CloudWatch will assume to deliver log events to the
+# Firehose delivery stream.
+resource "aws_iam_role" "cloudwatch_to_firehose" {
+  provider = aws.sharedservicesprovisionaccount
+
+  assume_role_policy = data.aws_iam_policy_document.cloudwatch_assume_role.json
+  description        = var.cloudwatch_to_firehose_role_description
+  name               = var.cloudwatch_to_firehose_role_name
+}
+
+# Create an IAM policy document that allows CloudWatch to write to the Firehose
+# delivery stream.
+data "aws_iam_policy_document" "cloudwatch_to_firehose" {
+  statement {
+    actions = [
+      "firehose:PutRecord",
+      "firehose:PutRecordBatch",
+    ]
+
+    effect = "Allow"
+
+    resources = [
+      "arn:aws:firehose:${var.aws_region}:${local.sharedservices_account_id}:deliverystream/${var.firehose_delivery_stream_name}",
```

Comment:
Are we unable to use `aws_kinesis_firehose_delivery_stream.cloudwatch_logs.arn`? I prefer not to handcraft ARNs unless specifically necessary.

Reply:
We cannot due to a circular dependency.  I added an explanatory comment in https://github.com/cisagov/cool-sharedservices-cdm/pull/46/commits/c3748705bf3573ee9af6b9ec58ed4c4c8f8a13ed.